app/components/submissions/fields/decision-documents-field.gjs

```diff
@@ -0,0 +1,276 @@
+      this.args.graphs.sourceGraph,
+    );
+  }
+
+  get path() {
+    return this.args.field.rdflibPath;
+  }
+
+  loadDocuments = task(async () => {
+    // This is needed so the rest of the method happens _after_ the willDestroy method of previous component instances,
+    // otherwise we would simply load the old data again before it was removed from the store.
+    await Promise.resolve();
+
+    const {
+      formStore,
+      sourceNode,
+      graphs: { sourceGraph },
+    } = this.args;
+
+    const triples = formStore.match(
+      sourceNode,
+      this.args.field.rdflibPath,
+      undefined,
+      sourceGraph,
+    );
+
+    const documentPromises = triples.map(async (triple) => {
+      const document = triple.object;
+
+      const url = new URL(
+        '/worship-decisions-cross-reference/document-information',
+        window.location.origin,
+      );
+
+      url.searchParams.append('forDecisionType', this.decisionType.uri);
+      url.searchParams.append('forRelatedDecision', document.uri);
+      const response = await fetch(url);
+
+      if (response.ok) {
+        const ttlData = await response.text();
+
+        if (ttlData.length > 0) {
+          let data = extractDocumentsFromTtl(ttlData);
+
+          return data.at(0);
+        }
+      } else {
+        throw new Error(
+          'Something went wrong while retrieving related document information',
+        );
+      }
+    });
+
+    this.documents = await Promise.all(documentPromises);
+  });
+
+  addDocuments = (documentsToAdd) => {
+    this.documents = [...this.documents, ...documentsToAdd];
+
+    const { formStore, graphs, sourceNode } = this.args;
+    const statements = documentsToAdd.map((document) => {
+      return {
+        subject: sourceNode,
+        predicate: this.path,
+        object: document.node,
+        graph: graphs.sourceGraph,
+      };
+    });
+
+    formStore.addAll(statements);
+
+    this.showModal = false;
+  };
+
+  removeDocument = (documentToRemove) => {
+    this.documents = this.documents.filter(
+      (document) => document !== documentToRemove,
+    );
+
+    const { formStore, graphs, sourceNode } = this.args;
+    formStore.removeStatements([
+      {
+        subject: sourceNode,
+        predicate: this.path,
+        object: documentToRemove.node,
+        graph: graphs.sourceGraph,
+      },
+    ]);
+  };
+
+  <template>
+    <AuLabel
+      @error={{this.hasErrors}}
+      @required={{this.isRequired}}
+      @warning={{this.hasWarnings}}
+      for={{this.inputId}}
+    >
+      {{@field.label}}
+    </AuLabel>
+
+    <AuHelpText @skin="secondary" class="au-u-margin-bottom-small">
+      <AuIcon @icon={{InfoCircleIcon}} />
+      Voor een vlotte raadpleging van de gerefereerde documenten, raden we aan
+      om eerst aan te loggen op
+      <AuLinkExternal
+        @skin="secondary"
+        href={{(worshipDecisionsDatabaseUrl)}}
+      >Databank Erediensten</AuLinkExternal>
+    </AuHelpText>
```

Comment:
Hmmmm, we probably don't want this in the version here since we're already in that app? I didn't think of that when I added that message to loket.

Reply:
Indeed, I'll comment it out !